### PR TITLE
Add sha224 RipeMd to unit test

### PR DESCRIPTION
**Pull request by jrblixt**
Adds Sha224 and RipeMd to the unit test application. Updates RipeMd return values and adds parameter checking where necessary.

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -3313,6 +3589,144 @@ static int test_wc_Sha384Final (void)
...
+    }
+
+    times = sizeof(hash_test) / sizeof(byte*);
+
+    /* Good test args. */
+    printf(testingFmt, "wc_sha224Final()");
+
+    for (i = 0; i < times; i++) {
+        if (!flag) {
+            ret = wc_Sha224Final(&sha224, hash_test[i]);
```
Comment (cconlon) [inline tests/api.c]: Is there a reason that there are three different output buffers (hash1, hash2, hash3), and that they are different sizes?  wc_Sha224Final() will always output SHA224_DIGEST_SIZE bytes into the 2nd argument.

File: tests/api.c [hunk 2]
```diff
@@ -3313,6 +3589,144 @@ static int test_wc_Sha384Final (void)
...
+    }
+
+    times = sizeof(hash_test) / sizeof(byte*);
+
+    /* Good test args. */
+    printf(testingFmt, "wc_RipeMdFinal()");
+
+    for (i = 0; i < times; i++) {
+        if (!flag) {
+            ret = wc_RipeMdFinal(&ripemd, hash_test[i]);
```
Comment (cconlon) [inline tests/api.c]: Is there a reason that there are three different output buffers (hash1, hash2, hash3), and that they are different sizes? wc_RipeMdFinal() will always output RIPEMD_DIGEST_SIZE bytes into the 2nd argument.

Comment (jrblixt) [inline tests/api.c]: The hash1, hash2, hash3 that are different sizes was designed and added to all of the hash algorithms as a white-box testing technique to test the input size and make sure that the library handled the larger buffer size appropriately and didn't cause a problem with the implementation.

**Comment by cconlon** [comment]
Thanks John!

**Pull request merged**